### DISCUSSION TITLE
Renovate will NOT create PR for patch-version-outdated packages

### DISCUSSION THREAD

**Discussion by ChiuMungZitAlexander** [renovatebot/renovate#21836] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Describe the problem
Configure renovate.json as below, especially `"bumpVersion": "patch"`
```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": ["config:base", ":dependencyDashboard"],
  "ignoreDeps": ["mongodb", "ethers"],
  "bumpVersion": "patch"
}
```

### Check if the renovate has already started a job - Yes
![image](https://user-images.githubusercontent.com/22234622/234771095-b4d0e854-f065-40ae-9b0c-eeaea795e4cf.png)

### Check if there's a patch version update for packages like swr - Yes
```
{
  "currentValue": "^2.1.3",
  "currentVersion": "2.1.5",
  "datasource": "npm",
  "depName": "swr",
  "depType": "dependencies",
  "homepage": "https://swr.vercel.app",
  "packageName": "swr",
  "prettyDepType": "dependency",
  "registryUrl": "https://registry.npmjs.org",
  "sourceUrl": "https://github.com/vercel/swr",
  "versioning": "npm",
  "warnings": [],
  "updates": []
}
```

### Check if renovate has created a PR? - No

repo
https://github.com/MorphisWallet/network

**Comment by rarkins** [comment]
Are you expecting your `package.json` to change, or your lock file only to change?

**Reply by ChiuMungZitAlexander** [reply]
Both, if possible.

Actually as you can see some of my packages are still outdated but renovate won't create PRs for them even though I set `"bumpVersion": "patch"`.
<img width="301" alt="image" src="https://user-images.githubusercontent.com/22234622/234776466-e9017c2c-1ce1-446c-ba0d-f38cbf67d606.png">

Anyway Renovate is still working, as below, for some packages.
<img width="535" alt="image" src="https://user-images.githubusercontent.com/22234622/234776356-dfae2083-8ec7-4dbc-950b-1fe189dfcfde.png">

**Reply by rarkins** [reply]
You are using ranges in your package.json so you won't get both unless the new value exceeds the current range

**Reply by ChiuMungZitAlexander** [reply]
Exactly, for some packages it is semantic versioning.
```
"dependencies": {
    "next": "^13.3.1",
    "swr": "^2.1.5",
  },
  // renovate will NOT update ⬆️
  // renovate will update ⬇️
  "devDependencies": {
    "@types/node": "18.16.1",
    "@types/react": "18.2.0",
    "eslint": "8.39.0",
    "eslint-config-next": "13.3.1",
    "typescript": "5.0.4"
  },
```

**Reply by rarkins** [reply]
Semantic Versioning only defines versions: https://semver.org/

npm and others have defined a syntax for version _constraints_, but such syntax is custom to the package manager (npm, pnpm, etc.) and is not officially part of semantic versioning. For example npm and composer both support "semantic versioning" but here are subtle differences to their constraint syntax.

Ultimately it doesn't really matter here because Renovate supports it, but it's incorrect to say that constraints means semantic versioning while versions do not, because the opposite is technically true.

**Reply by ChiuMungZitAlexander** [reply]
So I think we have a conclusion of this discussion.
